src/core/plugins/oas3/fn.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @prettier
+ */
+import { Map } from "immutable"
+
+export const isFileUploadIntended = (
+  schema,
+  { mediaType = null, fileUploadMediaTypes = [] } = {}
+) => {
+  const isFileUploadMediaType =
+    typeof mediaType === "string" &&
+    fileUploadMediaTypes.some((fileUploadMediaType) =>
+      mediaType.startsWith(fileUploadMediaType)
+    )
+
+  if (isFileUploadMediaType) {
+    return true
+  }
+
+  if (!Map.isMap(schema)) return false
```

Comment:
There is an option to make this code more flexible - meaning working both in Immutable and JavaScript realm.
When we detect the schema is immutable we adapt, when it's JavaScript object, we again adapt. We might avoid future failures when we pass schema as normal object.